components/service/experiments/src/test/java/mozilla/components/service/experiments/ExperimentEvaluatorTest.kt

```diff
@@ -251,6 +251,64 @@ class ExperimentEvaluatorTest {
         assertNull(evaluator.evaluate(context, ExperimentDescriptor("testexperiment"), listOf(experiment), 20))
     }
 
+    @Test
+    fun `evaluate appMinVersion`() {
+        testReset(appId = "test.appId", versionName = "1.0.0")
+
+        val experiment = createDefaultExperiment(
+            id = "testexperiment",
+            match = createDefaultMatcher(
+                localeLanguage = "eng",
+                appId = "test.appId",
+                regions = listOf("USA", "GBR"),
+                appMinVersion = "1.0.0",
+                deviceManufacturer = "unknown",
+                deviceModel = "robolectric",
+                localeCountry = "USA"
+            ),
+            buckets = Experiment.Buckets(20, 70),
+            lastModified = currentTime
+        )
+
+        val evaluator = ExperimentEvaluator()
+        assertNotNull(evaluator.evaluate(context, ExperimentDescriptor("testexperiment"), listOf(experiment), 20))
+
+        packageInfo.versionName = "1.1.18"
+        assertNotNull(evaluator.evaluate(context, ExperimentDescriptor("testexperiment"), listOf(experiment), 20))
+
+        packageInfo.versionName = "0.1.18"
+        assertNull(evaluator.evaluate(context, ExperimentDescriptor("testexperiment"), listOf(experiment), 20))
+    }
+
+    @Test
+    fun `evaluate appMaxVersion`() {
```

Comment:
Please also add a test for evaluating both min & max version together, both matching and non-matching.
Please also test on the edges of the possible value spaces, e.g. for `evaluate appMinVersion` with min=1.0.0, test for version being: 0.9.9, 1.0.0, 1.0.1, somewhere further away like 1.1.18.

Reply:
I expanded the test cases to better cover edge cases and close cases.